plugins/catalog-react/report-alpha.api.md

```diff
@@ -324,6 +321,54 @@ export interface EntityContentLayoutProps {
   }>;
 }
 
+// @alpha (undocumented)
+export type EntityPredicate =
+  | EntityPredicateExpression
+  | EntityPredicatePrimitive
+  | {
+      $all: EntityPredicate[];
+    }
+  | {
+      $any: EntityPredicate[];
+    }
+  | {
+      $not: EntityPredicate;
+    };
+
+// @alpha (undocumented)
+export type EntityPredicateExpression = {
+  [KPath in string]: EntityPredicateValue;
+} & {
+  [KPath in `$${string}`]: never;
+};
+
+// @alpha (undocumented)
+export type EntityPredicatePrimitive = string | number | boolean;
+
+// @alpha
+export function entityPredicateToFilterFunction<T extends JsonValue>(
```

Comment:
these probably should go into -common - especially the types - would it make sense to have them there from the start?

Reply:
Hmm, yep it does although at the same time it increases the exposure a bit. Moving the evaluation seems a bit tricky since I don't know if we really want people using that in the backend right now. On the other hand moving just the types doesn't seem very useful either. Thinking that it might be safest to keep it all in -react for now?